src/nvim/main.c

```diff
@@ -468,25 +461,12 @@ int main(int argc, char **argv)
     read_stdin();
   }
 
-  if (reading_tty && (need_wait_return || msg_didany)) {
-    // Because there's no UI yet, error messages would have been printed to
-    // stdout.  Before starting we need confirmation that the user has seen the
-    // messages and that is done with a call to wait_return.
-    TIME_MSG("waiting for return");
-    wait_return(true);
-  }
-
-  if (!headless_mode && !embedded_mode && !silent_mode) {
-    input_stop();  // Stop reading input, let the UI take over.
-    ui_builtin_start();
-  }
-
   setmouse();  // may start using the mouse
 
   if (exmode_active || early_ui) {
     // Don't clear the screen when starting in Ex mode, or when an
     // embedding UI might have displayed messages
-    must_redraw = CLEAR;
+    must_redraw = VALID;
```

Comment:
@bfredl tests pass with this change. Any objection?

Reply:
Let's change it. If there is startup redraw issues, we can change it back later...

Reply:
though it would be better then to have `redraw_later(VALID)` here in case something before this wants to redraw more than `VALID`.